core/src/main/scala/org/bitcoins/core/wallet/utxo/InputInfo.scala

```diff
@@ -20,6 +16,13 @@ import org.bitcoins.crypto.{ECPublicKey, NetworkElement, Sign}
 sealed trait InputInfo {
   def outPoint: TransactionOutPoint
 
+  def prevTransaction: Transaction
```

Comment:
I think that this belongs in `InputSigningInfo` rather than here because the previous transaction should not concern a party that isn't signing for this UTXO